File: mycollab-web/src/main/java/com/esofthead/mycollab/module/crm/view/setting/CrmSettingUrlResolver.java
package com.esofthead.mycollab.module.crm.view.setting;

import com.esofthead.mycollab.eventmanager.EventBus;
import com.esofthead.mycollab.module.crm.events.CrmSettingEvent;
import com.esofthead.mycollab.module.crm.view.CrmUrlResolver;

public class CrmSettingUrlResolver extends CrmUrlResolver {

	public CrmSettingUrlResolver() {
		this.addSubResolver("notification",
				new NotificationSettingUrlResolver());
		this.addSubResolver("customlayout", new CustomLayoutUrlResolver());
	}

	public static class NotificationSettingUrlResolver extends CrmUrlResolver {
		@Override
		protected void handlePage(String... params) {
			EventBus.getInstance().fireEvent(
					new CrmSettingEvent.GotoNotificationSetting(this, null));
		}
	}

	public static class CustomLayoutUrlResolver extends CrmUrlResolver {
		@Override
		protected void handlePage(String... params) {
			EventBus.getInstance().fireEvent(
					new CrmSettingEvent.GotoCustomViewSetting(this, null));
		}
	}
}

File: mycollab-web/src/main/scala/com/esofthead/mycollab/module/crm/view/setting/CrmSettingUrlResolver.scala
package com.esofthead.mycollab.module.crm.view.setting

import com.esofthead.mycollab.eventmanager.EventBusFactory
import com.esofthead.mycollab.module.crm.events.CrmSettingEvent
import com.esofthead.mycollab.module.crm.view.CrmUrlResolver

/**
  * @author MyCollab Ltd
  * @since 5.0.9
  */
class CrmSettingUrlResolver extends CrmUrlResolver {
  this.addSubResolver("notification", new NotificationSettingUrlResolver)
  this.addSubResolver("customlayout", new CustomLayoutUrlResolver)

  class NotificationSettingUrlResolver extends CrmUrlResolver {
    protected override def handlePage(params: String*) {
      EventBusFactory.getInstance().post(new CrmSettingEvent.GotoNotificationSetting(this, null))
    }
  }

  class CustomLayoutUrlResolver extends CrmUrlResolver {
    protected override def handlePage(params: String*) {
      EventBusFactory.getInstance().post(new CrmSettingEvent.GotoCustomViewSetting(this, null))
    }
  }

}
